NitroxClient/MonoBehaviours/ThrottledBuilder.cs

```diff
@@ -109,206 +98,29 @@ private void ActionBuildEvent(BuildEvent buildEvent)
             }
         }
 
-        private void PlaceBasePiece(BasePiecePlacedEvent basePiecePlacedBuildEvent)
+        private void ConstructionBegin(ConstructionBeginEvent constructionBegin)
         {
-            Log.Info("BuildBasePiece " + basePiecePlacedBuildEvent.BasePiece.Id + " type: " + basePiecePlacedBuildEvent.BasePiece.TechType + " parentId: " + basePiecePlacedBuildEvent.BasePiece.ParentId.OrElse(null));
-            BasePiece basePiece = basePiecePlacedBuildEvent.BasePiece;
-            GameObject buildPrefab = CraftData.GetBuildPrefab(basePiece.TechType.Enum());
-            MultiplayerBuilder.overridePosition = basePiece.ItemPosition;
-            MultiplayerBuilder.overrideQuaternion = basePiece.Rotation;
-            MultiplayerBuilder.overrideTransform = new GameObject().transform;
-            MultiplayerBuilder.overrideTransform.position = basePiece.CameraPosition;
-            MultiplayerBuilder.overrideTransform.rotation = basePiece.CameraRotation;
-            MultiplayerBuilder.placePosition = basePiece.ItemPosition;
-            MultiplayerBuilder.placeRotation = basePiece.Rotation;
-            MultiplayerBuilder.rotationMetadata = basePiece.RotationMetadata;
-            MultiplayerBuilder.Begin(buildPrefab);
-
-            GameObject parentBase = null;
-
-            if (basePiece.ParentId.HasValue)
-            {
-                parentBase = NitroxEntity.GetObjectFrom(basePiece.ParentId.Value).OrElse(null);
-            }
-            
-            Constructable constructable;
-            GameObject gameObject;
-
-            if (basePiece.IsFurniture)
-            {
-                SubRoot subRoot = (parentBase != null) ? parentBase.GetComponent<SubRoot>() : null;
-                                
-                gameObject = MultiplayerBuilder.TryPlaceFurniture(subRoot);
-                constructable = gameObject.RequireComponentInParent<Constructable>();
-            }
-            else
-            {
-                constructable = MultiplayerBuilder.TryPlaceBase(parentBase);
-                gameObject = constructable.gameObject;
-            }
-
-            NitroxEntity.SetNewId(gameObject, basePiece.Id);
-            
-            /**
-             * Manually call start to initialize the object as we may need to interact with it within the same frame.
-             */
-            MethodInfo startCrafting = typeof(Constructable).GetMethod("Start", BindingFlags.NonPublic | BindingFlags.Instance);
-            Validate.NotNull(startCrafting);
-            startCrafting.Invoke(constructable, new object[] { });
+            NitroxServiceLocator.LocateService<Building>().Constructable_ConstructionBegin_Remote(constructionBegin.BasePiece);
         }
 
         private void ConstructionCompleted(ConstructionCompletedEvent constructionCompleted)
```

Comment:
minor: we don't really need these anymore and can likely just move the one line up.